### PR TITLE
add the ability to wrap errors recovered from panics

### PR DESCRIPTION
**Pull request by nnchang**
This change is [Reviewable](https://reviewable.io/reviews/palantir/witchcraft-go-logging/43)

### CODE REVIEW

File: wlog/wapp/fatal.go [hunk 1]
```diff
@@ -32,13 +32,23 @@ func RunWithFatalLogging(ctx context.Context, runFn func(ctx context.Context) er
...
+			}
+		} else {
+			svc1log.FromContext(ctx).Error("panic recovered",
+				svc1log.SafeParam("stacktrace", stacktrace),
+				svc1log.UnsafeParam("recovered", r))
+			if retErr == nil {
+				retErr = werror.Error("panic recovered",
+					werror.SafeParam("stacktrace", stacktrace),
+					werror.UnsafeParam("recovered", r))
+			}
```
Comment (nnchang) [inline wlog/wapp/fatal.go:51]: couldn't come up with a prettier way to do this; feels like a good deal of repeated strings and such. Feedback welcome 🤷‍♂️

**Review by bmoylan — APPROVED** [review]
:lgtm:

> Reviewed 1 of 2 files at r1, 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/palantir/witchcraft-go-logging/43#-:-LcE0WNMEUia9vCBfGqQ:bnfp4nl)* status: all files reviewed, 1 unresolved discussion (waiting on @nnchang)

___
*[wlog/wapp/fatal.go, line 51 at r1](https://reviewable.io/reviews/palantir/witchcraft-go-logging/43#-LcDuyxqY6boq94iUgrx-r1-51:-LcE0AkSC0b4anpV1z8I:b-banfef) ([raw file](https://github.com/palantir/witchcraft-go-logging/blob/0aacb007ed66cbed9a4159da3cc2656558630b8f/wlog/wapp/fatal.go#L51)):*
<details><summary><i>Previously, nnchang (Nicholas Chang) wrote…</i></summary><blockquote>

couldn't come up with a prettier way to do this; feels like a good deal of repeated strings and such. Feedback welcome 🤷‍♂️ 
</blockquote></details>

yeah other than maybe moving the strings to constants I think this is fine

**Comment by nmiyake** [comment]
___
*[wlog/wapp/fatal.go, line 38 at r2](https://reviewable.io/reviews/palantir/witchcraft-go-logging/43#-Lgi5aJwEDBNpGiKDk_2:-Lgi5aJwEDBNpGiKDk_3:bkfuquc) ([raw file](https://github.com/palantir/witchcraft-go-logging/blob/149061d9bbd0b10bc8bef07d582449e02a8e020f/wlog/wapp/fatal.go#L38)):*
> ```Go
> 			svc1log.FromContext(ctx).Error("panic recovered",
> 				svc1log.SafeParam("stacktrace", stacktrace),
> 				svc1log.Stacktrace(err))
> ```

The `svc1log.Stacktrace` call also uses the key `stacktrace`, so it doesn't make sense to set both (since one will overwrite the other).

We should either use a different key for the manually added `stacktrace` or change the `svc1log.Stacktrace` call (pass in just the error as a string with a different key or something).

Should the `stacktrace` key be `threaddump` (or something similar) instead to differentiate it?

(just flagging this now since I ran into it in code I was running -- wanted to check the content of the error, but only had the thread dump due to this)

@bmoylan @nnchang

**Comment by bmoylan** [comment]
___
*[wlog/wapp/fatal.go, line 38 at r2](https://reviewable.io/reviews/palantir/witchcraft-go-logging/43#-Lgi5aJwEDBNpGiKDk_2:-LgiLawG15j9wBCXZ99X:bpgcpo9) ([raw file](https://github.com/palantir/witchcraft-go-logging/blob/149061d9bbd0b10bc8bef07d582449e02a8e020f/wlog/wapp/fatal.go#L38)):*
<details><summary><i>Previously, nmiyake (Nick Miyake) wrote…</i></summary><blockquote>

The `svc1log.Stacktrace` call also uses the key `stacktrace`, so it doesn't make sense to set both (since one will overwrite the other).

We should either use a different key for the manually added `stacktrace` or change the `svc1log.Stacktrace` call (pass in just the error as a string with a different key or something).

Should the `stacktrace` key be `threaddump` (or something similar) instead to differentiate it?

(just flagging this now since I ran into it in code I was running -- wanted to check the content of the error, but only had the thread dump due to this)

@bmoylan @nnchang 
</blockquote></details>

I don't mind if we change the param key, but they should be at different levels in the json (the params go into a `params` map, so this ends up at `params.stacktrace` rather than the top-level `stacktrace` field)

**Comment by nmiyake** [comment]
___
*[wlog/wapp/fatal.go, line 38 at r2](https://reviewable.io/reviews/palantir/witchcraft-go-logging/43#-Lgi5aJwEDBNpGiKDk_2:-Lgifsg85LlqLERkn7TT:bjrdda5) ([raw file](https://github.com/palantir/witchcraft-go-logging/blob/149061d9bbd0b10bc8bef07d582449e02a8e020f/wlog/wapp/fatal.go#L38)):*
<details><summary><i>Previously, bmoylan (Brad Moylan) wrote…</i></summary><blockquote>

I don't mind if we change the param key, but they should be at different levels in the json (the params go into a `params` map, so this ends up at `params.stacktrace` rather than the top-level `stacktrace` field)
</blockquote></details>

Oh right -- I must have missed that! Will take a closer look at my output to track down what's going on then. As long as the data is available, I have no issues with the key being the same (and agree it should be based on this).

**Pull request merged**